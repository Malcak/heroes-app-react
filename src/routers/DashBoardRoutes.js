import React from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import { Navbar } from '../components/ui/Navbar';
import { DCScreen } from '../components/dc/DCScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { HeroScreen } from '../components/heroes/HeroScreen';
import { SearchScreen } from '../components/search/SearchScreen';

export const DashBoardRoutes = () => {
  return (
    <>
      <Navbar></Navbar>
      <div className="container-fluid mt-2">
        <Switch>
          <Route exact path="/dc" component={DCScreen} />
          <Route exact path="/marvel" component={MarvelScreen} />
          <Route exact path="/hero/:heroId" component={HeroScreen} />
          <Route exact path="/search" component={SearchScreen} />
          <Redirect to="/dc" />
        </Switch>
      </div>
    </>
  );
};
