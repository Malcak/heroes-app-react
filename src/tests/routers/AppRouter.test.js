import React from 'react';
import { mount } from 'enzyme';
import { AuthContext } from '../../auth/AuthContext';
import { AppRouter } from '../../routers/AppRouter';

describe('test on the <AppRouter /> component', () => {
  const contextValue = {
    dispath: jest.fn(),
    user: {
      logged: false,
    },
  };

  test('should display the loginScreen if not authenticated', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('should display the <DCScreen /> if it is authenticated', () => {
    const contextValue = {
      dispath: jest.fn(),
      user: {
        logged: true,
        name: 'fernando',
      },
    };
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );
    expect(wrapper.find('.navbar').exists()).toBe(true);
  });
});
