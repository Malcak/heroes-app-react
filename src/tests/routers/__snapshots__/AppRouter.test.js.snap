// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on the <AppRouter /> component should display the loginScreen if not authenticated 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
